### PR TITLE
Add parsed functor material

### PR DESCRIPTION
**Pull request by GiudGiud**

### CODE REVIEW

File: framework/doc/content/source/materials/ParsedFunctorMaterial.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+Sets up a single functor material property that is computed using a parsed function expression.
+This is the equivalent of the [ParsedMaterial.md] but as a functor material property instead of a
+regular material property.
+
+A `ParsedFunctorMaterial` object takes the function expression as an input parameter in
+the form of a Function Parser expression. Parsed materials (unlike
+`ParsedFunctions`) can couple to nonlinear variables.
+In its configuration block all nonlinear variables the function depends on
+([!param](/Materials/ParsedFunctorMaterial/args)), as well as constants
```
Comment (dschwen) [inline framework/doc/content/source/materials/ParsedFunctorMaterial.md:13]:
```suggestion
([!param](/Materials/ParsedFunctorMaterial/variables)), as well as constants
```

**Comment by dschwen** [comment]
Please reduce copy/paste

File: framework/include/materials/ParsedFunctorMaterialHelper.h [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  using ParsedFunctorMaterialHelper<T>::_property;                                                        \
+  using ParsedFunctorMaterialHelper<T>::_nargs;
+
+
+/**
+ * Helper class to perform the parsing and optimization of the
+ * function expression.
+ */
+template <bool is_ad>
+class ParsedFunctorMaterialHelper : public FunctorMaterial, public FunctionParserUtils<is_ad>
```
[the remark this answers is not in the document]
Comment (GiudGiud) [reply framework/include/materials/ParsedFunctorMaterialHelper.h:38]: We use less things though, so we would not want to inherit the additional parameters and routines

[the remark this answers is not in the document]
Comment (dschwen) [reply framework/include/materials/ParsedFunctorMaterialHelper.h:38]: That's what inheritance can be used for

File: framework/include/materials/DerivativeMaterialInterface.h [hunk 3]
```diff
@@ -483,7 +483,7 @@ DerivativeMaterialInterface<T>::validateCoupling(const MaterialPropertyName & ba
 
     mooseWarning("Missing coupled variables {",
                  list,
-                 "} (add them to args parameter of ",
+                 "} (add them to the variables parameter of ",
```
Comment (GiudGiud) [inline framework/include/materials/DerivativeMaterialInterface.h:486]:
```suggestion
                 "} (add them to the coupled_variables parameter of ",
```

**Pull request closed without merging**